feat: Add Export Logs to Portal functionality

mrjeremystookey opened the pull request:
This commit introduces a new feature allowing users to export Ditto logs directly to the Portal.

The key changes include:
*   A new `ExportLogsToPortal` composable that displays an alert dialog to initiate the export process.
*   The composable triggers an update to the `__small_peer_info` collection to request the log upload.
*   A new screen and menu item, "Export Logs to Portal," are added to the main tools viewer.
*   The existing "Export Logs" label is clarified to "Export Logs to Device."

mrjeremystookey committed:
feat: Add Export Logs to Portal functionality

This commit introduces a new feature allowing users to export Ditto logs directly to the Portal.

The key changes include:
*   A new `ExportLogsToPortal` composable that displays an alert dialog to initiate the export process.
*   The composable triggers an update to the `__small_peer_info` collection to request the log upload.
*   A new screen and menu item, "Export Logs to Portal," are added to the main tools viewer.
*   The existing "Export Logs" label is clarified to "Export Logs to Device."

bplattenburg reviewed (commented):
Can we add a public method that customers can call that does this as well as the button? Not everyone will integrate this via our UI and giving them options helps

mrjeremystookey committed:
- removed delay

mrjeremystookey commented on DittoToolsAndroid/src/main/java/live/ditto/tools/exportlogs/ExportLogsToPortal.kt:
```diff
@@ -0,0 +1,107 @@
+                                    "currentTime" to currentTime, "peerKey" to peerKey
+                                )
+                            )
+                            delay(3000)
```
nope. Missed that. 
[- removed delay](https://github.com/getditto/DittoAndroidTools/pull/158/commits/62ca0110f7eb1eec3b2398f215e099809dfecb96)

mrjeremystookey replied:
and for sure. Give me a few on that public method.

rajramsaroop commented on DittoToolsAndroid/src/main/java/live/ditto/tools/exportlogs/ExportLogsToPortal.kt:
```diff
@@ -0,0 +1,106 @@
+            Text(title)
+        },
+        text = {
+            Text("Logs will be exported to Portal for appID: ${ditto.appId}")
```
We should make this text a string resource

mrjeremystookey committed:
extracted exporting... to string resource

mrjeremystookey committed:
extracted logs will be exported to string to string resource

mrjeremystookey committed:
added string resources

mrjeremystookey committed:
Adds a new public API, `DittoTools.requestLogExport()`, for programmatically triggering the export of device logs to the Ditto Portal.

This new function encapsulates the logic for updating the `__small_peer_info` collection, which was previously implemented directly in the `ExportLogsToPortal` composable. The composable is refactored to use this new API.

mrjeremystookey committed:
Refactors the "Export Logs to Portal" feature to use a ViewModel (`ExportLogsViewModel`) to manage its UI state.

This change introduces a sealed interface `ExportLogsUiState` to represent the various states of the log export process (Idle, Exporting, Success, Error).

The `ExportLogsToPortal` composable is updated to observe the state from the ViewModel, display loading indicators, handle errors gracefully by showing an error message and a "Retry" button, and show a success toast.

The function `requestLogExport` has been renamed to `uploadLogsToPortal` for clarity.

rajramsaroop reviewed (approved):
Nice work!

mrjeremystookey committed:
Refactor: Remove fully qualified name for MaterialTheme

The explicit `androidx.compose.material3.` qualifier for `MaterialTheme` is removed in favor of a direct import.

mrjeremystookey committed:
docs: Add Export Logs to Portal to README
